iroha_p2p/src/network.rs

```diff
@@ -246,27 +375,33 @@ where
         };
         match Peer::new_from(id.clone(), stream, self.broker.clone()) {
             Ok(peer) => {
+                let connection_id = peer.connection_id;
                 let peer = peer.start().await;
-                self.peers.insert(id, peer);
+                self.new_peers.insert(connection_id, peer.clone());
+                peer.do_send(Connect).await;
             }
             Err(e) => warn!(%e, "Unable to create peer"),
         }
     }
 }
 
-/// The message that is sent to [Network] to start connection to some other peer.
+/// The message that is sent to [`NetworkBase`] to start connection to some other peer.
 #[derive(Clone, Debug, iroha_actor::Message)]
-pub struct Connect {
+pub struct ConnectPeer {
```

Comment:
Do we need this suffix `Peer` here? Peers are the only thing to which network connects

Reply:
Yes, because we have `Connect` and `ConnectPeer` structures.

Reply:
Probably it would have been better to have `Connect` in public api and `ConnectPeer` in private

Reply:
I can renamed it like this: `ConnectPeer` -> `Connect` and `Connect` -> `Start`.

Reply:
Done this in that PR: https://github.com/hyperledger/iroha/pull/1350